diff --git a/dist/commonjs/registry.js b/dist/commonjs/registry.js
index d24d826852f3ceaabba44214120f4884c5e1ebff..15d8a6ca913f3ad262835a2e557d321e3478b937 100644
--- a/dist/commonjs/registry.js
+++ b/dist/commonjs/registry.js
@@ -592,7 +592,12 @@ function newField(proto, parentOrFile, reg, oneof, mapEntries) {
         field.oneof = undefined;
         field.typeName = typeName;
         field.jsonName = `[${typeName}]`; // option json_name is not allowed on extension fields
-        field.toString = () => `extension ${typeName}`;
+        Object.defineProperty(field, "toString", {
+            configurable: true,
+            enumerable: true,
+            value: () => `extension ${typeName}`,
+            writable: true,
+        });
         const extendee = reg.getMessage(trimLeadingDot(proto.extendee));
         assert(extendee, `invalid FieldDescriptorProto: extendee ${proto.extendee} not found`);
         field.extendee = extendee;
@@ -607,7 +612,12 @@ function newField(proto, parentOrFile, reg, oneof, mapEntries) {
             ? (0, names_js_1.protoCamelCase)(proto.name)
             : (0, names_js_1.safeObjectProperty)((0, names_js_1.protoCamelCase)(proto.name));
         field.jsonName = proto.jsonName;
-        field.toString = () => `field ${parent.typeName}.${proto.name}`;
+        Object.defineProperty(field, "toString", {
+            configurable: true,
+            enumerable: true,
+            value: () => `field ${parent.typeName}.${proto.name}`,
+            writable: true,
+        });
     }
     const label = proto.label;
     const type = proto.type;
diff --git a/dist/esm/registry.js b/dist/esm/registry.js
index fbd59ef17eee1662ea418d040fcddcf1227d8af1..0c0acb1c6d47335a805a9a042592a63f1f42e208 100644
--- a/dist/esm/registry.js
+++ b/dist/esm/registry.js
@@ -586,7 +586,12 @@ function newField(proto, parentOrFile, reg, oneof, mapEntries) {
         field.oneof = undefined;
         field.typeName = typeName;
         field.jsonName = `[${typeName}]`; // option json_name is not allowed on extension fields
-        field.toString = () => `extension ${typeName}`;
+        Object.defineProperty(field, "toString", {
+            configurable: true,
+            enumerable: true,
+            value: () => `extension ${typeName}`,
+            writable: true,
+        });
         const extendee = reg.getMessage(trimLeadingDot(proto.extendee));
         assert(extendee, `invalid FieldDescriptorProto: extendee ${proto.extendee} not found`);
         field.extendee = extendee;
@@ -601,7 +606,12 @@ function newField(proto, parentOrFile, reg, oneof, mapEntries) {
             ? protoCamelCase(proto.name)
             : safeObjectProperty(protoCamelCase(proto.name));
         field.jsonName = proto.jsonName;
-        field.toString = () => `field ${parent.typeName}.${proto.name}`;
+        Object.defineProperty(field, "toString", {
+            configurable: true,
+            enumerable: true,
+            value: () => `field ${parent.typeName}.${proto.name}`,
+            writable: true,
+        });
     }
     const label = proto.label;
     const type = proto.type;
